src/test/harness.rs

```diff
@@ -164,15 +158,29 @@ pub fn expect_messages(results: LsResultList, expected: &[&ExpectedMessage]) {
 }
 
 // Initialise the environment for a test.
-pub fn init_env(project_dir: &str) -> (Cache, TestCleanup) {
+pub fn init_env(project_dir: &str) -> (Config, Cache, TestCleanup) {
```

Comment:
I think it would be good to refactor the return type into a struct - config and cache could be fields and we could move the Drop impl from TestCleanup to the new Environment struct.

Reply:
Oh, and we could move mock_server and any similar functions to be methods on that object.

Reply:
Done.